### PR TITLE
Added Cloudflare DNS to yamls where missing

### LINKED ISSUE #209: cloudflare-dns not found error while sowing species gardener/virtual
**What happened**:
I am trying to deploy Gardener onto my cluster and things went well until `sowing species gardener/virtual` where I got the following error
```
2020/05/02 08:35:22 error generating manifest: unresolved nodes:
        (( *(.dns_credentials.[provider]) ))    in /mounted/home/sayak/landscape/crop/components/gardener/virtual/deployment.yaml       gardener.values.global.defaultDomains.[0].credentials  ()      *'cloudflare-dns' not found
        (( *(.dns_credentials.[provider]) ))    in /mounted/home/sayak/landscape/crop/components/gardener/virtual/deployment.yaml       gardener.values.global.internalDomain.credentials      ()      *'cloudflare-dns' not found
        (( *(.dns_credentials.[.landscape.dns.type]) )) in /mounted/home/sayak/landscape/crop/components/gardener/virtual/deployment.yaml       settings.dns_credentials       ()      *'cloudflare-dns' not found
error classification:
 *: error in local dynaml expression
 @: dependent of or involved in a cycle
 -: depending on a node with an error
Error: processing deployment.yaml
```
**What you expected to happen**:
I expected the cloudflare dns to work smoothly since it was added in 1.29.0
**How to reproduce it (as minimally and precisely as possible)**:
I think just trying to make a fresh deployment with cloudflare dns would trigger this
**Anything else we need to know?**:
I think I have somewhat found the issue and it is in https://github.com/gardener/garden-setup/blob/87b13681976fbf4ec92805fb4f4a632b50b3375b/components/gardener/virtual/deployment.yaml#L47 . Notice that the `dns_credentials` do not have `cloudflare-dns` as a key. I don't know if similar issues is present elsewhere.
**Environment**:

- Version of garden-setup 
1.31.0
- Versions of components (only needed if you overwrote the defaults)
- Where does the underlying base cluster come from and which operating system does it use?
AKS-Linux
- Which cloud provider is configured for the setup?
Azure
- Other:

### PR DESCRIPTION
**Pull request by SayakMukhopadhyay**
**What this PR does / why we need it**:
This PR fixes the absence of cloudflare-dns in many files where all other dns providers are present, including documentation
**Which issue(s) this PR fixes**:
Fixes #209 
**Special notes for your reviewer**:
Please verify this PR closely because I am not sure how everything works but I have still tried to make the changes as similar to existing code as possible.
**Release note**:

```improvement operator
Add missing support/documentation for cloudflare-dns
```

### CODE REVIEW

**Comment by SayakMukhopadhyay** [comment]
I am testing this and I have checked out my branch and have run `sow deploy -An` to resume but I am still getting the same error. I cannot even do a `sow delete all` or `sow delete -a gardener/virtual` as they are all giving the same error.

**Comment by Diaphteiros** [comment]
To be able to delete your landscape, you could manually delete the `gardener` folder from each of the `gen`, `export`, and `state` folders. This will make sow assume that the component is not deployed, so it will be skipped during deletion, which should circumvent that error. Since the component didn't get deployed (because the error already happened when rendering the manifests), there shouldn't be any infrastructure resources leaked using this workaround.

**Comment by SayakMukhopadhyay** [comment]
Thanks! I will try that. Anyway I am working on a test cluster so if that doesn't work I can just burn everything to the ground! 😁

**Comment by Diaphteiros** [comment]
Yes, but some components create stuff on the infrastructure that does not automatically disappear when deleting the base cluster, e.g. the `etcd/backupinfra` component creates an blobstore bucket. If you don't clean up the landscape 'the proper way', you will have to take care of those resources manually. Luckily, this is only the case for a few components :-)

**Comment by SayakMukhopadhyay** [comment]
Oh, that makes sense. Thanks for that info!

File: components/gardener/virtual/deployment.yaml [hunk 1]
```diff
@@ -69,6 +69,8 @@ dns_credentials:
     OS_DOMAIN_NAME: (( base64( landscape.dns.credentials.domainName ) ))
     OS_USER_DOMAIN_NAME: (( base64( landscape.dns.credentials.userDomainName || "" ) ))
     OS_PROJECT_NAME: (( base64( landscape.dns.credentials.tenantName ) ))
+  cloudflare-dns:
+    CLOUDFLARE_API_TOKEN: (( base64( landscape.dns.credentials.apiToken ) ))
```
Comment (Diaphteiros) [inline components/gardener/virtual/deployment.yaml:73]: This transformation of credentials is legacy code, from a time where the Gardener and the dns-controller-manager needed different keys for the credentials. I removed a similar mapping from the DNS controller component (see https://github.com/gardener/garden-setup/commit/8d14a033e8dbdab01c3c440304023e7c6fde2f7a), but I seem to have missed this one. 

I already created a PR to remove this mapping, I will merge it into master soon, so you can also remove this part (see https://github.com/gardener/garden-setup/pull/212).

**Review by Diaphteiros — CHANGES_REQUESTED** [review]
Thank you very much for adding the missing parts!
I added two requests for changes where you added something that should not be necessary anymore.

File: acre.yaml [hunk 2]
```diff
@@ -211,6 +211,7 @@ dns_type_mapping:
   aws: aws-route53
   azure: azure-dns
   openstack: openstack-designate
+  cloudflare: cloudflare-dns
```
[the remark this answers is not in the document]
Comment (SayakMukhopadhyay) [reply acre.yaml]: Got it!

File: components/gardener/virtual/deployment.yaml (hunk 1 above)
Comment (SayakMukhopadhyay) [reply components/gardener/virtual/deployment.yaml]: Reverted this change

**Comment by SayakMukhopadhyay** [comment]
I ran the `sow deploy` on a fresh cluster and the error is not happening anymore.

Note: I haven't finished the full deployment yet.

**Review by Diaphteiros — APPROVED** [review]
/lgtm

**Pull request merged**